lib/index.d.ts

```diff
@@ -0,0 +1,286 @@
+// Type definitions for stardog.js 0.3.1
+// Project: https://github.com/stardog-union/stardog.js
+// Definitions by: Stephen Nowell <http://github.com/snowell>, Ty Soehngen <http://github.com/tysoeh>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.1
+
+import Headers from 'fetch-ponyfill';
+import Guid from '@types/node';
+
+declare global {
+  
+  /** stardog.js: The Stardog JS API*/
+  namespace Stardog {
+
+    type ContentType =
+        'application/x-turtle' |
+        'text/turtle' |
+        'application/rdf+xml' |
+        'text/plain' |
+        'application/x-trig' |
+        'text/x-nquads' |
+        'application/trix';
+    
+    type MIMEType =
+        'text/plain' |
+        'application/json' |
+        'text/boolean';
+
+    interface HTTPOptions {
+        headers?: Headers;
+        method?: string;
+        mimeType?: MIMEType;
+        contentType?: ContentType;
+    }
+
+    interface QueryOptions {
+        query: string;
+        baseURI?: string;
+        limit?: number;
+        offset?: number;
+        reasoning?: boolean;
+    }
+
+    interface HTTPMessage {
+        status: string;
+        statusText: string;
+    }
+
+    interface HTTPBody extends HTTPMessage, result {}
+
+    interface result {
+        result: string;
+    }
+
+    interface Params {
+        [paramName: string]: any;
+    }
+    
+    interface ConnectionOptions {
+        endpoint: string;
+        username: string;
+        password: string;
+    }
+
+    /** Describes the connection to a running Stardog server. */
+    class Connection {
+        constructor(ConnectionOptions);
+
+        config(options: ConnectionOptions): void;
+        headers(): Headers;
+        uri(resource: Array<string>): string;
+    }
+    
+    namespace server {
+        /** Shuts down a Stardog server. */
+        function shutdown(conn: Connection, params?: Params) : Promise<HTTPMessage>;
+    }
+
+    /** Stardog database actions. */
+    namespace db {
+
+        interface DatabaseOptions {
```

Comment:
You're not going to be able to type them like this. For the sake of maintenance, I'd just let this be an object and move on. The database object looks like https://github.com/stardog-union/stardog.js/blob/develop/lib/db/dbopts.js